--- a/include/pci_ids/radeonsi_pci_ids.h	2026-10-08 23:26:12.161089681 +0300
+++ b/include/pci_ids/radeonsi_pci_ids.h	2026-10-08 23:26:12.606854700 +0300
@@ -122,6 +122,11 @@
 CHIPSET(0x985E, KABINI)
 CHIPSET(0x985F, KABINI)
 
+CHIPSET(0x9920, LIVERPOOL)
+CHIPSET(0x9922, LIVERPOOL)
+CHIPSET(0x9923, LIVERPOOL)
+CHIPSET(0x9924, GLADIUS)
+
 CHIPSET(0x1304, KAVERI)
 CHIPSET(0x1305, KAVERI)
 CHIPSET(0x1306, KAVERI)
--- a/src/amd/addrlib/src/amdgpu_asic_addr.h	2026-10-08 23:26:12.162836523 +0300
+++ b/src/amd/addrlib/src/amdgpu_asic_addr.h	2026-10-08 23:26:12.607125274 +0300
@@ -63,7 +63,9 @@
 #define AMDGPU_HAWAII_RANGE     0x28, 0x3C //# 40 <= x < 60
 
 #define AMDGPU_SPECTRE_RANGE    0x01, 0x41 //#   1 <= x < 65
-#define AMDGPU_SPOOKY_RANGE     0x41, 0x81 //#  65 <= x < 129
+#define AMDGPU_SPOOKY_RANGE     0x41, 0x61
+#define AMDGPU_STARSHA_RANGE    0x61, 0x71
+#define AMDGPU_STARSHP_RANGE    0x71, 0x81
 #define AMDGPU_KALINDI_RANGE    0x81, 0xA1 //# 129 <= x < 161
 #define AMDGPU_GODAVARI_RANGE   0xA1, 0xFF //# 161 <= x < max
 
@@ -140,6 +142,8 @@
 
 #define ASICREV_IS_SPECTRE(r)          ASICREV_IS(r, SPECTRE)
 #define ASICREV_IS_SPOOKY(r)           ASICREV_IS(r, SPOOKY)
+#define ASICREV_IS_STARSHA(r)          ASICREV_IS(r, STARSHA)
+#define ASICREV_IS_STARSHP(r)          ASICREV_IS(r, STARSHP)
 #define ASICREV_IS_KALINDI(r)          ASICREV_IS(r, KALINDI)
 #define ASICREV_IS_KALINDI_GODAVARI(r) ASICREV_IS(r, GODAVARI)
 
--- a/src/amd/common/ac_gpu_info.c	2026-10-08 23:26:12.166098616 +0300
+++ b/src/amd/common/ac_gpu_info.c	2026-10-08 23:26:12.637879971 +0300
@@ -616,6 +616,8 @@
       case CHIP_VERDE:
       case CHIP_HAINAN:
       case CHIP_BONAIRE:
+      case CHIP_LIVERPOOL:
+      case CHIP_GLADIUS:
       case CHIP_KAVERI:
       case CHIP_ICELAND:
       case CHIP_CARRIZO:
@@ -670,6 +672,8 @@
       case FAMILY_KV:
          identify_chip2(SPECTRE, KAVERI);
          identify_chip2(SPOOKY, KAVERI);
+         identify_chip2(STARSHA, LIVERPOOL);
+         identify_chip2(STARSHP, GLADIUS);
          identify_chip2(KALINDI, KABINI);
          identify_chip2(GODAVARI, KABINI);
          break;
@@ -2246,6 +2250,8 @@
    case CHIP_TAHITI:
    case CHIP_PITCAIRN:
    case CHIP_VERDE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_BONAIRE:
    case CHIP_HAWAII:
    case CHIP_TONGA:
@@ -2316,6 +2322,14 @@
       raster_config = 0x3a00161a;
       raster_config_1 = 0x0000002e;
       break;
+   case CHIP_LIVERPOOL:
+      raster_config = 0x2a00161a;
+      raster_config_1 = 0x00000000;
+      break;
+   case CHIP_GLADIUS:
+      raster_config = 0x2a00161a;
+      raster_config_1 = 0x0000002e;
+      break;
    default:
       fprintf(stderr, "ac: Unknown GPU, using 0 for raster_config\n");
       raster_config = 0x00000000;
--- a/src/amd/common/amd_family.c	2026-10-08 23:26:12.166295133 +0300
+++ b/src/amd/common/amd_family.c	2026-10-08 23:26:12.608058584 +0300
@@ -18,6 +18,8 @@
    CASE(OLAND);
    CASE(HAINAN);
    CASE(BONAIRE);
+   CASE(LIVERPOOL);
+   CASE(GLADIUS);
    CASE(KABINI);
    CASE(KAVERI);
    CASE(HAWAII);
@@ -93,6 +95,10 @@
       return GFX8;
    if (family >= CHIP_BONAIRE)
       return GFX7;
+   if (family >= CHIP_LIVERPOOL)
+      return GFX7;
+   if (family >= CHIP_GLADIUS)
+      return GFX7;
 
    return GFX6;
 }
@@ -112,6 +118,10 @@
       return "hainan";
    case CHIP_BONAIRE:
       return "bonaire";
+   case CHIP_LIVERPOOL:
+      return "bonaire";
+   case CHIP_GLADIUS:
+      return "bonaire";
    case CHIP_KABINI:
       return "kabini";
    case CHIP_KAVERI:
--- a/src/amd/common/amd_family.h	2026-10-08 23:26:12.166295133 +0300
+++ b/src/amd/common/amd_family.h	2026-10-08 23:26:12.608232998 +0300
@@ -79,6 +79,8 @@
    CHIP_OLAND,
    CHIP_HAINAN,
    /* GFX7 (Sea Islands) */
+   CHIP_LIVERPOOL,
+   CHIP_GLADIUS,
    CHIP_BONAIRE,
    CHIP_KAVERI,
    CHIP_KABINI,
--- a/src/amd/vulkan/meta/radv_meta_fast_clear.c	2026-10-08 23:26:12.176712031 +0300
+++ b/src/amd/vulkan/meta/radv_meta_fast_clear.c	2026-10-08 23:26:12.608464610 +0300
@@ -354,7 +354,7 @@
 
    switch (op) {
    case FAST_CLEAR_ELIMINATE:
-      pred_offset = image->fce_pred_offset;
+      pred_offset = 0;
       break;
    case FMASK_DECOMPRESS:
       pred_offset = 0; /* FMASK_DECOMPRESS is never predicated. */
--- a/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c	2026-10-08 23:26:12.313680490 +0300
+++ b/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c	2026-10-08 23:26:12.608775232 +0300
@@ -329,6 +329,8 @@
       ws->info.gfx_level = GFX6;
       break;
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_KAVERI:
    case CHIP_KABINI:
    case CHIP_HAWAII:
@@ -542,6 +544,8 @@
    case CHIP_VERDE:
    case CHIP_OLAND:
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_KAVERI:
       ws->info.max_tcc_blocks = 4;
       break;
@@ -571,9 +575,11 @@
       case CHIP_TAHITI:
       case CHIP_PITCAIRN:
       case CHIP_BONAIRE:
+      case CHIP_LIVERPOOL:
          ws->info.max_se = 2;
          break;
       case CHIP_HAWAII:
+      case CHIP_GLADIUS:
          ws->info.max_se = 4;
          break;
       }
@@ -667,6 +673,8 @@
    case CHIP_VERDE:
    case CHIP_HAINAN:
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_KAVERI:
       ws->info.l2_cache_size = ws->info.num_tcc_blocks * 128 * 1024;
       break;
--- a/src/gallium/winsys/radeon/drm/radeon_surface.c	2026-10-08 23:26:12.313680490 +0300
+++ b/src/gallium/winsys/radeon/drm/radeon_surface.c	2026-10-08 23:26:12.609124128 +0300
@@ -77,6 +77,8 @@
     CHIP_OLAND,
     CHIP_HAINAN,
     CHIP_BONAIRE,
+    CHIP_LIVERPOOL,
+    CHIP_GLADIUS,
     CHIP_KAVERI,
     CHIP_KABINI,
     CHIP_HAWAII,
